x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts

```diff
@@ -0,0 +1,203 @@
+  SearchResponse,
+} from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { MetricThresholdParams } from '@kbn/infra-plugin/common/alerting/metrics';
+import { ThresholdParams } from '@kbn/observability-plugin/common/custom_threshold_rule/types';
+import { RoleCredentials } from '@kbn/ftr-common-functional-services';
+import { DeploymentAgnosticFtrProviderContext } from '../ftr_provider_context';
+
+export interface SloBurnRateRuleParams {
+  sloId: string;
+  windows: WindowSchema[];
+  dependencies?: Dependency[];
+}
+
+interface WindowSchema {
+  id: string;
+  burnRateThreshold: number;
+  maxBurnRateThreshold: number;
+  longWindow: Duration;
+  shortWindow: Duration;
+  actionGroup: string;
+}
+
+interface Dependency {
+  ruleId: string;
+  actionGroupsToSuppressOn: string[];
+}
+
+type DurationUnit = 'm' | 'h' | 'd' | 'w' | 'M';
+
+interface Duration {
+  value: number;
+  unit: DurationUnit;
+}
+
+export function AlertingApiProvider({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const samlAuth = getService('samlAuth');
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const es = getService('es');
+  const retry = getService('retry');
+  const logger = getService('log');
+  const config = getService('config');
+  const retryTimeout = config.get('timeouts.try');
+  const requestTimeout = 30 * 1000;
+
+  return {
+    async waitForRuleStatus({
+      ruleId,
+      expectedStatus,
+      roleAuthc,
+    }: {
+      ruleId: string;
+      expectedStatus: string;
+      roleAuthc: RoleCredentials;
+    }) {
+      return await retry.tryForTime(retryTimeout, async () => {
+        const response = await supertestWithoutAuth
+          .get(`/api/alerting/rule/${ruleId}`)
+          .set(roleAuthc.apiKeyHeader)
+          .set(samlAuth.getInternalRequestHeader())
+          .timeout(requestTimeout);
+        const { execution_status: executionStatus } = response.body || {};
+        const { status } = executionStatus || {};
+        if (status !== expectedStatus) {
+          throw new Error(`waitForStatus(${expectedStatus}): got ${status}`);
+        }
+        return executionStatus?.status;
+      });
+    },
+
+    async waitForDocumentInIndex<T>({
+      indexName,
+      docCountTarget = 1,
+    }: {
+      indexName: string;
+      docCountTarget?: number;
+    }): Promise<SearchResponse<T, Record<string, AggregationsAggregate>>> {
+      return await retry.tryForTime(retryTimeout, async () => {
+        const response = await es.search<T>({
+          index: indexName,
+          rest_total_hits_as_int: true,
+        });
+        logger.debug(`Found ${response.hits.total} docs, looking for atleast ${docCountTarget}.`);
+
+        if (
+          !response.hits.total ||
+          (typeof response.hits.total === 'number'
+            ? response.hits.total < docCountTarget
+            : response.hits.total.value < docCountTarget)
+        ) {
+          throw new Error('No hits found');
```

Comment:
Another nit: technically, `No hits found` does not cover all the cases. It could also be that there are hits found but just not enough, i.e. less than `docCountTarget`, so this error message might potentially be confusing when debugging.

Reply:
@pheyos would that work? `throw new Error('No hits or not enough hits found')`

Otherwise we would need to print different message in both cases.

Reply:
@mgiota yes, I think that would work.